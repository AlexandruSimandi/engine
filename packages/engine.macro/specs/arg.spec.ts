import pluginTester from "babel-plugin-tester";
import prettier from "prettier";
import plugin from "babel-plugin-macros";

const macroFile = "'@c11/engine.macro'";

pluginTester({
  plugin,
  babelOptions: { filename: __filename },
  formatResult: (result: any) => {
    return prettier.format(result, {
      parser: "babel",
    });
  },
  tests: {
    "should support Arg": {
      code: `
        import { producer } from ${macroFile}
        const result = producer((
          a1 = '123',
          a2 = Arg.a1,
          a3 = Arg.a2[Arg.a1],
          a4 = Arg.a3[Prop.foo],
        ) => { })
      `,
      snapshot: true,
    },
  },
});
