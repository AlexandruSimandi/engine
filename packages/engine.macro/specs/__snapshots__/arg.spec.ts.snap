// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros should support Arg: should support Arg 1`] = `

import { producer } from '@c11/engine.macro'
const result: producer = ({
  a1 = '123',
  a2 = Arg.a1,
  a3 = Arg.a2[Arg.a1],
  a4 = Arg.a3[Prop.foo],
}) => { }

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

var result = {
  args: {
    type: "STRUCT",
    value: {
      a1: {
        type: "VALUE",
        value: {
          type: "CONST",
          value: "123",
        },
      },
      a2: {
        type: "VALUE",
        value: {
          type: "INTERNAL",
          path: ["a1"],
        },
      },
      a3: {
        type: "VALUE",
        value: {
          type: "INTERNAL",
          path: ["a2", "$a1"],
        },
      },
      a4: {
        type: "VALUE",
        value: {
          type: "INTERNAL",
          path: ["a3", "@foo"],
        },
      },
    },
    meta: {},
  },
  fn: function fn(_ref) {
    var a1 = _ref.a1,
      a2 = _ref.a2,
      a3 = _ref.a3,
      a4 = _ref.a4;
  },
};


`;
